### PR TITLE
Add View command & abstract some context stuff out

### PR DESCRIPTION
**Pull request by podocarp**
Issue #50.
Depends on PR #104.

### CODE REVIEW

**Comment by podocarp** [comment]
I'm going to hold off adding tests since we are gonna change the command really soon anyways and deprecate this one.

File: src/main/java/seedu/address/model/Model.java [hunk 1]
```diff
@@ -144,6 +144,13 @@
      */
     void updateFilteredPersonList(Predicate<Person> predicate);
 
+    /**
+     * Updates the filter of the filtered person list to filter by the given {@code predicate},
+     * and also sets the context to point at the new filtered list.
+     * @throws NullPointerException if {@code predicate} is null.
+     */
+    void updateFilteredPersonList(Predicate<Person> predicate, Model model);
```
Comment (liakify) [inline src/main/java/seedu/address/model/Model.java]: Why does this thing need a Model object?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -213,12 +213,30 @@ public void setActivity(Activity target, Activity editedActivity) {
...
+    }
+
     @Override
     public void updateFilteredPersonList(Predicate<Person> predicate) {
         requireNonNull(predicate);
         filteredPersons.setPredicate(predicate);
     }
 
+    @Override
+    public void updateFilteredPersonList(Predicate<Person> predicate, Model model) {
```
Comment (liakify) [inline src/main/java/seedu/address/model/ModelManager.java:235]: Why are you allowing model objects to be passed in? I though just `this.setContext()` is enough...?

File: src/main/java/seedu/address/model/Model.java (hunk 1 above)
Comment (liakify) [inline src/main/java/seedu/address/model/Model.java:152]: Why does this thing need a Model object?

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                        String.format(Messages.MESSAGE_ACTIVITY_NONEXISTENT, id));
+            }
+            predicate = (Person x) -> a.hasPerson(x.getPrimaryKey());
+        } else if (prefix.equals(PREFIX_CONTACT_ID)) {
+            predicate = (Person x) -> x.getPrimaryKey() == id;
+        } else {
+            predicate = x -> true; // default to showing everyone if no args
+        }
+
+        model.updateFilteredPersonList(predicate, model);
```
Comment (liakify) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java:53]: This is just a placeholder for viewing activities right? Like our actual activity viewing logic isn't going to be like this

**Review by Aulud — COMMENTED** [review]
Will deprecate this but implement some of the changes

**Pull request closed without merging**